### ISSUE TITLE
magit-commit not working (again) under cygwin 

### ISSUE THREAD

**Issue by philippmoeller** [magit/magit#3491]
In a git repository I try to commit like this:

    M-x magit-status
    c c

and the response in the magit-process buffer is:

```
  1 C:/cygwin64/usr/libexec/git-core/git.exe … commit --

*ERROR*: Setting current directory: Permission denied, c:/cygdrive/c/my-repo/.git/error: There was a problem with the editor 'c\:/emacs/bin/emacsclient.exe'.
Please supply the message using either -m or -F option.
```
Note the extra `cygdrive/c/` part in the path. The seems to be the same issue as in #1318 but none of the workarounds there seem to work for me, hence the new issue. 

The same happens when trying to interactively rebase as well.

I'm using:

    M-x magit-version
    Magit 20180618.1934, Git 2.17.0, Emacs 25.2.1, windows-nt
    M-x emacs-version
    GNU Emacs 25.2.1 (x86_64-w64-mingw32) of 2017-04-24

**Comment by npostavs** [comment]
Can you post the output of `M-x with-editor-debug`, and `M-: (magit-expand-git-file-name "/cygdrive/c/my-repo/.git/")`,

And the output when you evaluate this (make sure to have something staged):
```elisp
(with-current-buffer (get-buffer-create "*magit #3491 test*")
  (erase-buffer)
  (with-editor
    (let ((magit-git-debug t)
          (process-environment
           (cons "GIT_TRACE=1"
                 (cons "GIT_EDITOR=printf '[%s]\n' \"$EDITOR\"" process-environment))))
      (magit-git-insert "commit")))
  (display-buffer (current-buffer)))
```

**Comment by npostavs** [comment]
> `[c\:/emacs/bin/emacsclient.exe]`

Aha, what's that backslash doing there? This looks like #2794, although that one is about Git for Windows with `shell-file-name` set to cygwin `bash`. Maybe it's something new that cygwin git also expects w32 quoting?

**Comment by mgrunwald** [comment]
Hello,

I have the same problem :(

```
with-editor: c:/Users/m.grunwald/Documents/.emacs.d/elpa/with-editor-20180618.1602/with-editor.el

emacs: c:/Program Files (x86)/emacs/bin/emacs (25.2.1)

system:

  system-type: windows-nt

  system-configuration: x86_64-w64-mingw32

  system-configuration-options: --without-dbus --without-compress-install 'CFLAGS=-O2 -static -g3'

server:

  server-running-p: t

  server-process: #<process server>

  server-use-tcp: t

  server-name: server

  server-socket-dir: nil

    WARNING: not an accessible directory

  server-auth-dir: ~\.emacs.d\server\

    server

with-editor-emacsclient-executable:

value:   c:/Program Files (x86)/emacs/bin/emacsclient.exe (25.2)

default: c:/Program Files (x86)/emacs/bin/emacsclient.exe (25.2)

funcall: c:/Program Files (x86)/emacs/bin/emacsclient.exe (25.2)

path:

  $PATH: "C:\\cygwin32\\bin;C:\\ProgramData\\Anaconda3;C:\\ProgramData\\Anaconda3\\Library\\usr\\bin;C:\\ProgramData\\Anaconda3\\Library\\bin;C:\\ProgramData\\Anaconda3\\Scripts;C:\\Program Files (x86)\\Silicon Labs\\ISA3 Utilities\\bin;C:\\ProgramData\\Oracle\\Java\\javapath;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\Program Files (x86)\\Enterprise Vault\\EVClient\\;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\Program Files\\Microsoft SQL Server\\110\\Tools\\Binn\\;C:\\Program Files\\doxygen\\bin;c:\\Program Files (x86)\\IAR Systems\\Embedded Workbench 7.2\\common\\bin;c:\\Program Files\\Tools\\;c:\\Program Files\\OmniSharp\\;C:\\Program Files\\cURL\\bin;c:\\Program Files\\cURL\\bin\\;C:\\Program Files\\Microsoft SQL Server\\120\\Tools\\Binn\\;C:\\Program Files (x86)\\Windows Kits\\10\\Windows Performance Toolkit\\;c:\\Program Files (x86)\\gnu-global\\bin\\;C:\\Program Files (x86)\\GNU Tools ARM Embedded\\4.9 2015q3\\bin;c:\\Program Files (x86)\\emacs\\bin\\;c:\\Program Files\\TortoiseSVN\\bin\\;C:\\Program Files (x86)\\PDFtk Server\\bin\\;C:\\Program Files (x86)\\Pico Technology\\PicoScope6\\;C:\\Program Files\\MiKTeX 2.9\\miktex\\bin\\x64\\;C:\\Program Files\\WIDCOMM\\Bluetooth Software\\;C:\\Program Files\\WIDCOMM\\Bluetooth Software\\syswow64;C:\\Program Files\\TortoiseSVN\\bin;C:\\Program Files\\Git\\cmd;C:\\Program Files\\Calibre2\\;C:\\Program Files\\LLVM\\bin;c:\\Program Files (x86)\\Graphviz2.38\\bin\\;C:\\Program Files\\MatLABR2015b\\runtime\\win64;C:\\Program Files\\MatLABR2015b\\bin;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\Program Files (x86)\\gnuplot\\bin;"

  exec-path: (c:/cygwin32/bin C:/ProgramData/Anaconda3 C:/ProgramData/Anaconda3/Library/usr/bin C:/ProgramData/Anaconda3/Library/bin C:/ProgramData/Anaconda3/Scripts C:/Program Files (x86)/Silicon Labs/ISA3 Utilities/bin C:/ProgramData/Oracle/Java/javapath C:/WINDOWS/system32 C:/WINDOWS C:/WINDOWS/System32/Wbem C:/Program Files (x86)/Enterprise Vault/EVClient/ C:/WINDOWS/System32/WindowsPowerShell/v1.0/ C:/WINDOWS/System32/WindowsPowerShell/v1.0/ C:/Program Files/Microsoft SQL Server/110/Tools/Binn/ C:/Program Files/doxygen/bin c:/Program Files (x86)/IAR Systems/Embedded Workbench 7.2/common/bin c:/Program Files/Tools/ c:/Program Files/OmniSharp/ C:/Program Files/cURL/bin c:/Program Files/cURL/bin/ C:/Program Files/Microsoft SQL Server/120/Tools/Binn/ C:/Program Files (x86)/Windows Kits/10/Windows Performance Toolkit/ c:/Program Files (x86)/gnu-global/bin/ C:/Program Files (x86)/GNU Tools ARM Embedded/4.9 2015q3/bin c:/Program Files (x86)/emacs/bin/ c:/Program Files/TortoiseSVN/bin/ C:/Program Files (x86)/PDFtk Server/bin/ C:/Program Files (x86)/Pico Technology/PicoScope6/ C:/Program Files/MiKTeX 2.9/miktex/bin/x64/ C:/Program Files/WIDCOMM/Bluetooth Software/ C:/Program Files/WIDCOMM/Bluetooth Software/syswow64 C:/Program Files/TortoiseSVN/bin C:/Program Files/Git/cmd C:/Program Files/Calibre2/ C:/Program Files/LLVM/bin c:/Program Files (x86)/Graphviz2.38/bin/ C:/Program Files/MatLABR2015b/runtime/win64 C:/Program Files/MatLABR2015b/bin C:/WINDOWS/System32/WindowsPowerShell/v1.0/ C:/Program Files (x86)/gnuplot/bin . c:/Program Files (x86)/emacs/libexec/emacs/25.2/x86_64-w64-mingw32)

  with-editor-emacsclient-path:

    c:/Program Files (x86)/emacs/bin (t)

      c:/Program Files (x86)/emacs/bin/emacsclient.exe (25.2)

    c:/cygwin32/bin (t)

    C:/ProgramData/Anaconda3 (t)

    C:/ProgramData/Anaconda3/Library/usr/bin (nil)

    C:/ProgramData/Anaconda3/Library/bin (t)

    C:/ProgramData/Anaconda3/Scripts (t)

    C:/Program Files (x86)/Silicon Labs/ISA3 Utilities/bin (t)

    C:/ProgramData/Oracle/Java/javapath (t)

    C:/WINDOWS/system32 (t)

    C:/WINDOWS (t)

    C:/WINDOWS/System32/Wbem (t)

    C:/Program Files (x86)/Enterprise Vault/EVClient/ (t)

    C:/Program Files/Microsoft SQL Server/110/Tools/Binn/ (t)

    C:/Program Files/doxygen/bin (t)

    c:/Program Files (x86)/IAR Systems/Embedded Workbench 7.2/common/bin (t)

    c:/Program Files/Tools/ (t)

    c:/Program Files/OmniSharp/ (t)

    C:/Program Files/cURL/bin (t)

    c:/Program Files/cURL/bin/ (t)

    C:/Program Files/Microsoft SQL Server/120/Tools/Binn/ (t)

    C:/Program Files (x86)/Windows Kits/10/Windows Performance Toolkit/ (t)

    c:/Program Files (x86)/gnu-global/bin/ (t)

    C:/Program Files (x86)/GNU Tools ARM Embedded/4.9 2015q3/bin (t)

    c:/Program Files (x86)/emacs/bin/ (t)

      c:/Program Files (x86)/emacs/bin/emacsclient.exe (25.2)

    c:/Program Files/TortoiseSVN/bin/ (t)

    C:/Program Files (x86)/PDFtk Server/bin/ (t)

    C:/Program Files (x86)/Pico Technology/PicoScope6/ (t)

    C:/Program Files/MiKTeX 2.9/miktex/bin/x64/ (nil)

    C:/Program Files/WIDCOMM/Bluetooth Software/ (t)

    C:/Program Files/WIDCOMM/Bluetooth Software/syswow64 (t)

    C:/Program Files/TortoiseSVN/bin (t)

    C:/Program Files/Git/cmd (t)

    C:/Program Files/Calibre2/ (t)

    C:/Program Files/LLVM/bin (t)

    c:/Program Files (x86)/Graphviz2.38/bin/ (t)

    C:/Program Files/MatLABR2015b/runtime/win64 (nil)

    C:/Program Files/MatLABR2015b/bin (nil)

    C:/WINDOWS/System32/WindowsPowerShell/v1.0/ (t)

    C:/Program Files (x86)/gnuplot/bin (t)

    . (t)

    c:/Program Files (x86)/emacs/libexec/emacs/25.2/x86_64-w64-mingw32 (t)

 

 

(magit-expand-git-file-name "/cygdrive/c/Users/m.grunwald/Documents/Projects/LUXeye/ESME_box_tests/testingenvironment/.git")

"c:/Users/m.grunwald/Documents/Projects/LUXeye/ESME_box_tests/testingenvironment/.git"

 

 

["c:/Program Files (x86)/emacs/bin/emacsclient.exe"]

[/cygdrive/c/Users/m.grunwald/Documents/Projects/LUXeye/ESME_box_tests/testingenvironment/.git/COMMIT_EDITMSG]
```

**Comment by philippmoeller** [comment]
@npostavs `shell-file-name` is

> shell-file-name is a variable defined in ‘C source code’.
> Its value is "bash"
> Original value was 
> "C:/msys64/home/user/emacs-git/build/emacs-25.2/x86_64/nt/cmdproxy.exe"

Unfortunately I cannot find anything in #2794 that helps me to resolve this issue. Any ideas?

**Comment by philippmoeller** [comment]
Thanks for the tip, but it did not fix the issue.

The error message is still the same:

> *ERROR*: Setting current directory: Permission denied, c:/cygdrive/c/gitrepos/my-repo/.git/error: There was a problem with the editor '"c:/emacs/bin/emacsclient.exe"'.

Notice how the `/cygdrive/c` should not be in there.

**Comment by npostavs** [comment]
Hmm, not sure what's going on then. Could you try debugging `git-commit-setup` to see why it's not doing the right thing? This initial expression is supposed to handle exactly that case:

```elisp
(defun git-commit-setup ()
  ;; cygwin git will pass a cygwin path (/cygdrive/c/foo/.git/...),
  ;; try to handle this in window-nt Emacs.
  (--when-let
      (and (eq system-type 'windows-nt)
           (not (file-accessible-directory-p default-directory))
           (if (require 'magit-git nil t)
               ;; Emacs prepends a "c:".
               (magit-expand-git-file-name (substring buffer-file-name 2))
             ;; Fallback if we can't load `magit-git'.
             (and (string-match "\\`[a-z]:/\\(cygdrive/\\)?\\([a-z]\\)/\\(.*\\)"
                                buffer-file-name)
                  (concat (match-string 2 buffer-file-name) ":/"
                          (match-string 3 buffer-file-name)))))
    (when (file-accessible-directory-p (file-name-directory it))
      (find-alternate-file it)))
  [...]
```

**Comment by philippmoeller** [comment]
I used `debug-on-entry git-commit-setup` but this function was never entered.

**Comment by npostavs** [comment]
Oh, hmm. The only place it's called from is `git-commit-setup-check-buffer`, which should be in `find-file-hook`. Can you check those?

**Comment by npostavs** [comment]
Okay, instead of poking around with the debugger, let's cast a wider net with tracing:

```elisp
(dolist (fun '(find-file-noselect
	       find-file-noselect-1
	       after-find-file
	       git-commit-setup-check-buffer
	       git-commit-setup
	       find-alternate-file
	       server-visit-files
	       server-execute))
  (trace-function fun nil (lambda () (symbol-value 'find-file-hooks))))
```

**Comment by philippmoeller** [comment]
Trying to commit with `c c` resulted in this:

> 1 -> (server-execute #<process server6152 <127.0.0.1:65534>> (("c:/cygdrive/c/gitrepos/test-project/.git/COMMIT_EDITMSG")) nil nil t nil nil)(mode-local-post-major-mode-change magit-edit-branch*description-check-buffers global-magit-file-mode-check-buffers magit-auto-revert-mode-check-buffers #[0 \302\301!\210\303\304!8\211\207 [buffer-file-name auto-revert-tail-pos make-local-variable 7 file-attributes] 3] git-commit-setup-check-buffer global-font-lock-mode-check-buffers epa-file-find-file-hook vc-refresh-state which-func-ff-hook)
> | 2 -> (server-visit-files (("c:/cygdrive/c/gitrepos/test-project/.git/COMMIT_EDITMSG")) #<process server6152 <127.0.0.1:65534>> nil)(mode-local-post-major-mode-change magit-edit-branch*description-check-buffers global-magit-file-mode-check-buffers magit-auto-revert-mode-check-buffers #[0 \302\301!\210\303\304!8\211\207 [buffer-file-name auto-revert-tail-pos make-local-variable 7 file-attributes] 3] git-commit-setup-check-buffer global-font-lock-mode-check-buffers epa-file-find-file-hook vc-refresh-state which-func-ff-hook)
> | 2 <- server-visit-files: (#<buffer COMMIT_EDITMSG>)(mode-local-post-major-mode-change magit-edit-branch*description-check-buffers global-magit-file-mode-check-buffers magit-auto-revert-mode-check-buffers #[0 \302\301!\210\303\304!8\211\207 [buffer-file-name auto-revert-tail-pos make-local-variable 7 file-attributes] 3] git-commit-setup-check-buffer global-font-lock-mode-check-buffers epa-file-find-file-hook vc-refresh-state which-func-ff-hook)
> 1 <- server-execute: nil(mode-local-post-major-mode-change magit-edit-branch*description-check-buffers global-magit-file-mode-check-buffers magit-auto-revert-mode-check-buffers #[0 \302\301!\210\303\304!8\211\207 [buffer-file-name auto-revert-tail-pos make-local-variable 7 file-attributes] 3] git-commit-setup-check-buffer global-font-lock-mode-check-buffers epa-file-find-file-hook vc-refresh-state which-func-ff-hook)

**Comment by npostavs** [comment]
Do you have an empty `COMMIT_EDITMSG` buffer hanging around? Try killing it before committing.

**Comment by philippmoeller** [comment]
Ah, yes. The first time around a buffer is created. This time after a fresh start:

> 1 -> (server-execute #<process server1672 <127.0.0.1:50168>> (("c:/cygdrive/c/gitrepos/test-project/.git/COMMIT_EDITMSG")) nil nil t nil nil)(magit-edit-branch*description-check-buffers global-magit-file-mode-check-buffers magit-auto-revert-mode-check-buffers #[0 \302\301!\210\303\304!8\211\207 [buffer-file-name auto-revert-tail-pos make-local-variable 7 file-attributes] 3] git-commit-setup-check-buffer global-font-lock-mode-check-buffers epa-file-find-file-hook vc-refresh-state which-func-ff-hook)
> | 2 -> (server-visit-files (("c:/cygdrive/c/gitrepos/test-project/.git/COMMIT_EDITMSG")) #<process server1672 <127.0.0.1:50168>> nil)(magit-edit-branch*description-check-buffers global-magit-file-mode-check-buffers magit-auto-revert-mode-check-buffers #[0 \302\301!\210\303\304!8\211\207 [buffer-file-name auto-revert-tail-pos make-local-variable 7 file-attributes] 3] git-commit-setup-check-buffer global-font-lock-mode-check-buffers epa-file-find-file-hook vc-refresh-state which-func-ff-hook)
> | | 3 -> (find-file-noselect "c:/cygdrive/c/gitrepos/test-project/.git/COMMIT_EDITMSG")(magit-edit-branch*description-check-buffers global-magit-file-mode-check-buffers magit-auto-revert-mode-check-buffers #[0 \302\301!\210\303\304!8\211\207 [buffer-file-name auto-revert-tail-pos make-local-variable 7 file-attributes] 3] git-commit-setup-check-buffer global-font-lock-mode-check-buffers epa-file-find-file-hook vc-refresh-state which-func-ff-hook)
> | | | 4 -> (find-file-noselect-1 #<buffer COMMIT_EDITMSG> "c:/cygdrive/c/gitrepos/test-project/.git/COMMIT_EDITMSG" nil nil "c:/cygdrive/c/gitrepos/test-project/.git/COMMIT_EDITMSG" nil)(magit-edit-branch*description-check-buffers global-magit-file-mode-check-buffers magit-auto-revert-mode-check-buffers #[0 \302\301!\210\303\304!8\211\207 [buffer-file-name auto-revert-tail-pos make-local-variable 7 file-attributes] 3] git-commit-setup-check-buffer global-font-lock-mode-check-buffers epa-file-find-file-hook vc-refresh-state which-func-ff-hook)
> | | | | 5 -> (after-find-file t t)(magit-edit-branch*description-check-buffers global-magit-file-mode-check-buffers magit-auto-revert-mode-check-buffers #[0 \302\301!\210\303\304!8\211\207 [buffer-file-name auto-revert-tail-pos make-local-variable 7 file-attributes] 3] git-commit-setup-check-buffer global-font-lock-mode-check-buffers epa-file-find-file-hook vc-refresh-state which-func-ff-hook)
> | | | | 5 <- after-find-file: !non-local\ exit!(magit-edit-branch*description-check-buffers global-magit-file-mode-check-buffers magit-auto-revert-mode-check-buffers #[0 \302\301!\210\303\304!8\211\207 [buffer-file-name auto-revert-tail-pos make-local-variable 7 file-attributes] 3] git-commit-setup-check-buffer global-font-lock-mode-check-buffers epa-file-find-file-hook vc-refresh-state which-func-ff-hook)
> | | | 4 <- find-file-noselect-1: !non-local\ exit!(magit-edit-branch*description-check-buffers global-magit-file-mode-check-buffers magit-auto-revert-mode-check-buffers #[0 \302\301!\210\303\304!8\211\207 [buffer-file-name auto-revert-tail-pos make-local-variable 7 file-attributes] 3] git-commit-setup-check-buffer global-font-lock-mode-check-buffers epa-file-find-file-hook vc-refresh-state which-func-ff-hook)
> | | 3 <- find-file-noselect: !non-local\ exit!(magit-edit-branch*description-check-buffers global-magit-file-mode-check-buffers magit-auto-revert-mode-check-buffers #[0 \302\301!\210\303\304!8\211\207 [buffer-file-name auto-revert-tail-pos make-local-variable 7 file-attributes] 3] git-commit-setup-check-buffer global-font-lock-mode-check-buffers epa-file-find-file-hook vc-refresh-state which-func-ff-hook)
> | 2 <- server-visit-files: !non-local\ exit!(magit-edit-branch*description-check-buffers global-magit-file-mode-check-buffers magit-auto-revert-mode-check-buffers #[0 \302\301!\210\303\304!8\211\207 [buffer-file-name auto-revert-tail-pos make-local-variable 7 file-attributes] 3] git-commit-setup-check-buffer global-font-lock-mode-check-buffers epa-file-find-file-hook vc-refresh-state which-func-ff-hook)
> 1 <- server-execute: nil(magit-edit-branch*description-check-buffers global-magit-file-mode-check-buffers magit-auto-revert-mode-check-buffers #[0 \302\301!\210\303\304!8\211\207 [buffer-file-name auto-revert-tail-pos make-local-variable 7 file-attributes] 3] git-commit-setup-check-buffer global-font-lock-mode-check-buffers epa-file-find-file-hook vc-refresh-state which-func-ff-hook)
>

**Comment by npostavs** [comment]
Okay, does `(setq debug-on-signal t debug-on-error t)` get you a backtrace? (make sure the `COMMIT_EDITMSG` buffer is deleted too)

**Comment by philippmoeller** [comment]
Yes.

[stacktrace.txt](https://github.com/magit/magit/files/2159398/stacktrace.txt)

**Comment by npostavs** [comment]
Oh, that's the wrong one.  Can you hit `c` (`debugger-continue`) in the `*Backtrace*` buffer until you get a backtrace with `after-find-file` in it?

**Comment by philippmoeller** [comment]
The next two:

[stacktrace2.txt](https://github.com/magit/magit/files/2159431/stacktrace2.txt)

**Comment by npostavs** [comment]
Okay, I think see the problem now, please try the patch in #3505.

**Comment by philippmoeller** [comment]
No change. The path still contains the extra `/cygdrive/c/`.

**Comment by mgrunwald** [comment]
Am 4. Juli 2018 09:00:05 MESZ schrieb philippmoeller <notifications@github.com>:
>No change. The path still contains the extra `/cygdrive/c/`.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/magit/magit/issues/3491#issuecomment-402382941

Hi, for me, the patch worked!
Thank you 😀
-- 
Markus Grunwald

**Comment by npostavs** [comment]
Okay, this is interesting. @philippmoeller do you get the same backtraces with `after-find-file`, or are they different?

Does the trace output look different?

**Comment by tarsius** [comment]
@philippmoeller ping!

**Comment by philippmoeller** [comment]
@npostavs Sorry, for the delay. Here is the stacktrace with the patch applied:

> Debugger entered--Lisp error: (file-error "Setting current directory" "Permission denied" "c:/cygdrive/c/gitrepos/test/.git/")
>   call-process("git" nil #<buffer  *temp*> nil "config" "core.commentchar")
>   apply(call-process "git" nil #<buffer  *temp*> nil ("config" "core.commentchar"))
>   process-lines("git" "config" "core.commentchar")
>   (car (process-lines "git" "config" "core.commentchar"))
>   (progn (car (process-lines "git" "config" "core.commentchar")))
>   (condition-case nil (progn (car (process-lines "git" "config" "core.commentchar"))) (error nil))
>   (or (condition-case nil (progn (car (process-lines "git" "config" "core.commentchar"))) (error nil)) "#")
>   (set (make-local-variable (quote comment-start)) (or (condition-case nil (progn (car (process-lines "git" "config" "core.commentchar"))) (error nil)) "#"))
>   git-commit-setup-font-lock()
>   (and buffer-file-name (string-match-p git-commit-filename-regexp buffer-file-name) (git-commit-setup-font-lock))
>   git-commit-setup-font-lock-in-buffer()
>   run-hooks(after-change-major-mode-hook)
>   run-mode-hooks()
>   fundamental-mode()
>   normal-mode(t)
>   after-find-file(t t)
>   find-file-noselect-1(#<buffer COMMIT_EDITMSG> "c:/cygdrive/c/gitrepos/test/.git/COMMIT_EDITMSG" nil nil "c:/cygdrive/c/gitrepos/cosmos-jenkins-library/.git/COMMIT_EDITMSG" nil)
>   find-file-noselect("c:/cygdrive/c/gitrepos/test/.git/COMMIT_EDITMSG")
>   #[770 "\305\306r\211\203\205

**Comment by npostavs** [comment]
@philippmoeller can you try again with `debug-on-signal` set back to `nil`.

**Comment by npostavs** [comment]
Hmm, for some reason `fix #xxx` syntax didn't close this when I merged #3505, perhaps Github is confused by the `maint` branch. Anyway, I think this should be fixed now, please post back if it's not.
